async-nats/src/lib.rs

```diff
@@ -493,29 +493,50 @@ struct Subscription {
 #[derive(Debug)]
 struct SubscriptionContext {
     next_sid: u64,
+    next_uid: u64,
     subscription_map: HashMap<u64, Subscription>,
+    uid_map: HashMap<u64, u64>,
 }
 
 impl SubscriptionContext {
-    pub fn new() -> SubscriptionContext {
+    pub(crate) fn new() -> SubscriptionContext {
```

Comment:
Don't pub crate methods like this, the type is internal and that's enough.

```suggestion
    pub fn new() -> SubscriptionContext {
```